awx/api/serializers.py

```diff
@@ -1105,6 +1104,20 @@ def create(self, validated_data):
             )
         return obj
         
+        
+class OAuth2TokenDetailSerializer(OAuth2TokenSerializer):
+
+    refresh_token = serializers.SerializerMethodField()
+    token = serializers.SerializerMethodField()
+
+    class Meta:
+        model = OAuth2AccessToken
+        fields = (
+            '*', '-name', 'description', 'user', 'token', 'refresh_token',
+            'application', 'expires', 'scope',
+        )
+        read_only_fields = ('user', 'token', 'expires', 'application')       
```

Comment:
try using only

```python
read_only_fields = ('*', 'application', 'user')
```

The star should inherit fields from the parent classes via the metaclass.

I don't know why 'token' or 'expires' would have ever been writable in the first place. And I'm not sure what we're doing with that user field.

Reply:
That is much cleaner.  But wouldn't this make `description` immutable as well?